### DISCUSSION TITLE
Compiling gunrock to a shared library

### DISCUSSION THREAD

**Discussion by dma-neves** [gunrock/gunrock#1060] [Q&A] [answered]
Hello! I'm trying to compile Gunrock to a shared library so I can easily use it in a set of custom benchmarks I'm developing. The Compile instructions (I'm referring to these: https://gunrock.github.io/docs/#/gunrock/building_gunrock?id=compilation) suggest that GUNROCK_BUILD_SHARED_LIBS is set to ON by default. But regardless of setting this flag ON or leaving the default value, I can't seem to get it to compile the share libs. Whenever I try to set the flag manually I also get "Manually-specified variables were not used by the project: GUNROCK_BUILD_SHARED_LIB". Any help would be appreciated, thank you.

**Comment by neoblizz** [comment]
Hi @dma-neves, apologies, that is legacy documentation that is still online (updated: https://gunrock.github.io/gunrock/). We've converted Gunrock to be header only, you would have to modify cmake a bit to get it to build as a shared-library (I have not experimented with this.)

https://github.com/gunrock/gunrock/blob/main/CMakeLists.txt#L98-L99
I think this will have to be modified to `SHARED` to build shared library.

Some docs: https://stackoverflow.com/questions/17511496/how-to-create-a-shared-library-with-cmake

**Reply by dma-neves** [reply, marked as answer]
No problem. I managed to get it to compile to shared library but then had to fix a bunch of linking problems and eventually gave up. So I ended up just using a CMakeLists that adds gunrock as a sub_directory. Not sure if it's the best solution but it gets the job done. Thanks for the help!
